lib/langchain/conversation.rb

```diff
@@ -84,11 +85,44 @@ def examples
 
     private
 
-    def llm_response(prompt)
-      @llm.chat(messages: @memory.messages, context: @memory.context, examples: @memory.examples, **@options, &@block)
+    def call_llm
+      @llm.chat(messages: transform_messages(messages), context: context&.to_s, examples: transform_messages(examples), **@options, &@block)
     rescue Langchain::Utils::TokenLength::TokenLimitExceeded => exception
       @memory.reduce_messages(exception)
       retry
     end
+
+    def completion_response_path
+      key = @block ? "STREAMING_RESPONSE_PATH" : "COMPLETION_RESPONSE_PATH"
+
+      safe_const_get(llm_klass, key)
+    end
+
+    def transform_messages(messages)
```

Comment:
If we really want to have the LLMs know nothing about other stuff, the `transform_messages` method needs to be moved to the Conversation, so that LLMs already receive messages in the corresponding format.